src/main/java/org/springframework/data/elasticsearch/core/mapping/ElasticsearchPersistentEntity.java

```diff
@@ -49,4 +51,22 @@
 	String settingPath();
 
 	boolean isCreateIndexAndMapping();
+
+	/**
+	 * Returns whether the {@link ElasticsearchPersistentEntity} has an score property. If this call returns
+	 * {@literal true}, {@link #getScoreProperty()} will return a non-{@literal null} value.
+	 *
+	 * @return false when {@link ElasticsearchPersistentEntity} does not define a score property.
```

Comment:
Let's add `@since 3.1` for newly introduced API. Probably also for the method below, and `ScoredPage` etc.